### PR TITLE
Fix Docker for Mac issue with `--service-token` (close #1649)

### LINKED ISSUE #1649: Scope fails to launch on latest Docker for Mac (beta18)
```
> bash ./scope launch
68d432bf5c07d17e7b42204c85bddc1a9d0ac30869324329645d094357019d0d
docker: Error response from daemon: Mounts denied: 
The path /var/run/scope/plugins
is not shared from OS X and does not belong to the system.
You can configure shared paths from Docker -> Preferences...
..
```

### PR DESCRIPTION
**Pull request by errordeveloper**

### CODE REVIEW

File: scope [hunk 1]
```diff
@@ -195,6 +195,7 @@ EOF
         if ! check_docker_for_mac || check_probe_only ; then
             check_not_running $SCOPE_CONTAINER_NAME $SCOPE_IMAGE_NAME
             docker rm -f $SCOPE_CONTAINER_NAME >/dev/null 2>&1 || true
+            create_plugins_dir
```
Comment (errordeveloper) [inline scope]: We could make this `check_docker_for_mac && create_plugins_dir`, but I'm not sure if we really have to...

Comment (rade) [inline scope]: This ends up creating the plugin dir even when not on a Mac. Which I thought we tried to avoid.

Comment (rade) [inline scope]: > We could make this `check_docker_for_mac && create_plugins_dir`

Why not put this before the 'if'? Then we don't need `create_plugins_dir` in two branches.

**Comment by 2opremio** [comment]
The launch code could really use a refactoring. We check for `check_docker_for_mac` at 3 different places and `check_probe_only` at 2 different places.

**Comment by 2opremio** [comment]
@jml Can you please take a look at the refactoring I made?

File: scope [hunk 2]
```diff
@@ -192,41 +206,36 @@ EOF
...
             # - https://forums.docker.com/t/ports-in-host-network-namespace-are-not-accessible/10789
             check_not_running $SCOPE_APP_CONTAINER_NAME $SCOPE_IMAGE_NAME
             docker rm -f $SCOPE_APP_CONTAINER_NAME >/dev/null 2>&1 || true
-            CONTAINER=$($(launch_app_command "$@"))
+            CONTAINER=$($(launch_docker4mac_app_command "$@"))
             echo $CONTAINER
             app_ip=$(docker inspect -f '{{.NetworkSettings.IPAddress}}' "${CONTAINER}")
             check_not_running $SCOPE_CONTAINER_NAME $SCOPE_IMAGE_NAME
             docker rm -f $SCOPE_CONTAINER_NAME >/dev/null 2>&1 || true
             create_plugins_dir
```
Comment (errordeveloper) [inline scope]: Is this redundant now?

File: scope [hunk 3]
```diff
@@ -146,13 +146,27 @@ launch_command() {
...
 
-launch_app_command() {
+launch_docker4mac_app_command() {
     echo docker run -d --name=$SCOPE_APP_CONTAINER_NAME \
             -e CHECKPOINT_DISABLE \
             -p 0.0.0.0:4040:4040 \
             $WEAVESCOPE_DOCKER_ARGS $SCOPE_IMAGE --no-probe "$@"
 }
 
+launch_normally() {
```
Comment (jml) [inline scope]: Please add a documentation comment: "normally" is a bit of a weasel word.

File: scope (hunk 2 above)
Comment (2opremio) [inline scope]: Oh, yeah! Good catch!

File: scope (hunk 3 above)
Comment (2opremio) [inline scope]: Would a better name help instead? `do_default_launch`?

Comment (2opremio) [inline scope]: OK, that name is not better, but I take suggestions ...

Comment (rade) [inline scope]: `launch`

Comment (jml) [inline scope]: I think adding a comment describing what it's _not_ would help.

**Comment by errordeveloper** [comment]
I'm looking at this now, and I see that `create_plugins_dir` and some other functions are not consistent with other function. I've spoke with @tomwilkie, and the reason `launch_command` does `echo docker run ...` and is later called from a subshell is because subshell creates new scope, and thereby prevents crazy use of global variables etc. I am wondering whether we might want to ensure consistency, once we are refactoring this...

File: scope [hunk 4]
```diff
@@ -146,13 +146,27 @@ launch_command() {
...
+    check_not_running $SCOPE_CONTAINER_NAME $SCOPE_IMAGE_NAME
+    docker rm -f $SCOPE_CONTAINER_NAME >/dev/null 2>&1 || true
+    CONTAINER=$($(launch_command "$@"))
+    echo $CONTAINER    
+}
+
+print_app_endpoints() {
+    echo "Weave Scope is reachable at the following URL(s):" >&2
+    for ip in $@; do
```
Comment (jml) [inline scope]: Quotes around `$@`, e.g. `"$@"`.

File: scope (hunk 3 above)
Comment (2opremio) [inline scope]: @rade suggested `launch()` let's go for that.

**Comment by jml** [comment]
Thanks. Refactoring LGTM.

**Comment by 2opremio** [comment]
> create_plugins_dir` and some other functions are not consistent with other function

Let's not fix the world in this PR :) I only did the refactoring because it was easier than understanding the code for reviewing the PR.

**Comment by errordeveloper** [comment]
Sadly, I'm getting this on Docker for Mac:
```
0 %> ./scope launch --service-token=MYTOKEN
d6fa231c17f8ce7d669d8f92015f100d7fffc16a7e42161c1b9176b68c87892e
./scope: line 213: return: can only `return' from a function or sourced script

1 %> ./scope launch
af3b8d723dab193662c60c5ea96e2e379de4342c1e5cf249ddb2781e5ea5286b
weavescope is already running.

1 %> ./scope stop

0 %> ./scope launch
c9dea490b5f8cb391168d05531e4a80ebdea08987a5858456a5fdcd92be5bf08
Weave Scope is reachable at the following URL(s):
  * http://localhost:4040/
./scope: line 229: return: can only `return' from a function or sourced script

1 %> ./scope stop

0 %> ./scope launch --service-token=MYTOKEN
c011d7673d9fb774f4aca16c93acd506694e5c863bafe305edf124c58dca4a4e
./scope: line 213: return: can only `return' from a function or sourced script

1 %> 
```

**Comment by errordeveloper** [comment]
Tested on Docker for Mac, all works well. LGTM.

**Pull request merged**